### ISSUE TITLE
pytorch bug?

### ISSUE THREAD

**Issue by lzmchina** [bytedeco/javacpp-presets#1250]
I want to use `adaptive_avg_pool2d` function to transform input tensor to output tensor:
```java
output = adaptive_avg_pool2d(input, new LongArrayRef(new long[] { 7, 7 }, 2));
```
The input variable's shape is `[64, 512, 7, 7]`, but the output's shape is  `[64, 512, 0, 7]`.

The correct output shape should be `[64, 512, 7, 7]`

**Comment by saudet** [comment]
You're probably not calling the function you want to call. Try to call the one with AdaptiveAvgPool2dOptions as parameter.

**Comment by lzmchina** [comment]
> You're probably not calling the function you want to call. Try to call the one with AdaptiveAvgPool2dOptions as parameter.

I have tried it at the first, but I still got a wrong shape tensor.

**Comment by saudet** [comment]
If you have a working example in C++, please provide it. It should be easy to translate it to Java.

**Comment by saudet** [comment]
Ah, I see what's wrong. We need an array of LongOptional for this. I've pushed a fix in commit https://github.com/bytedeco/javacpp-presets/commit/94268bb30cbe3ed3e53d204eaebadeb5b20115af, and it's a bit awkward, but it works this way:
```java
LongOptionalVector size = new LongOptionalVector(new LongOptional(7), new LongOptional(7));
Tensor output = adaptive_avg_pool2d(input, size.get(0));
```
Thanks for reporting!

**Comment by saudet** [comment]
The fix for this has been released with version 1.5.8! Enjoy

**Comment by mullerhai** [comment]
> et(0)



> Ah, I see what's wrong. We need an array of LongOptional for this. I've pushed a fix in commit [94268bb](https://github.com/bytedeco/javacpp-presets/commit/94268bb30cbe3ed3e53d204eaebadeb5b20115af), and it's a bit awkward, but it works this way:
> 
> LongOptionalVector size = new LongOptionalVector(new LongOptional(7), new LongOptional(7));
> Tensor output = adaptive_avg_pool2d(input, size.get(0));
> Thanks for reporting!

there only get(0) , if it will bring three elements ? so  get(1) get(2)  not work?

**Comment by mullerhai** [comment]
Hi ,@saudet @HGuillemet    and  FractionalMaxPool2d FractionalMaxPool3d ,need pass three parameter Instance ,contain [ kernel_size,   output_size  output_ratio]
but FractionalMaxPool2dImpl FractionalMaxPool3d  structor can not pass three get(0)
how to do think  below code ,and them can not compile , all wrong 
```
FractionalMaxPool2dImpl( kernel_size.get(0) , output_size.get(0),  output_ratio.get(0) ] 
FractionalMaxPool3dImpl( kernel_size.get(0) , output_size.get(0),  output_ratio.get(0) ]
```
so let us  make FractionalMaxPool2dImpl  FractionalMaxPool3d   Options  could work  maybe  will solve them bugs
